### PR TITLE
Rails 4.0.4 fix

### PR DESCRIPTION
**Pull request by tiegz**
### WHAT

A small fix to support Rails 4.0.4+
### WHY

There was a commit (https://github.com/rails/rails/commit/714634ad02b443ab51f8ef3ded324de411715d2a) that removed a method that `sematnic-attributes` calls.

### CODE REVIEW

**Comment by tiegz** [comment]
doh, just realized my branch was off of `cainlevy`'s version, which seems different than `kickstarter`'s.

**Comment by tiegz** [comment]
or rather, my branch was off of `kickstarter` (which is the link from rubygems page), but the current version seems to only be in `cainlevy`.

File: lib/semantic_attributes/attribute_formats.rb [hunk 1]
```diff
@@ -44,12 +44,17 @@ def humanize(attr, obj)
...
         self.semantic_attributes[attr].predicates.inject(obj) { |val, predicate| val = predicate.normalize(val) }
       end
-      
+
       protected
 
       def define_normalization_method_for(attr)
-        self.define_attribute_methods if self.respond_to? :attribute_methods_generated? and !self.attribute_methods_generated?
-        
+        if ::ActiveRecord::VERSION::STRING >= "4.0.4"
```
Comment (ktheory) [inline lib/semantic_attributes/attribute_formats.rb]: This would break on v4.0.10, right?

File: Gemfile.lock [hunk 2]
```diff
@@ -1,92 +1,87 @@
 PATH
   remote: .
   specs:
-    semantic_attributes (1.0.1)
```
Comment (ktheory) [inline Gemfile.lock:4]: Should `Gemfile.lock` be git ignored for a library?

File: lib/semantic_attributes/attribute_formats.rb (hunk 1 above)
Comment (tiegz) [inline lib/semantic_attributes/attribute_formats.rb]: doh! Thanks for the reminder, such an annoying gotcha.

File: Gemfile.lock (hunk 2 above)
Comment (tiegz) [inline Gemfile.lock:4]: Yeah, but that seems out of this PR's scope. I could do another commit that removes it.

File: lib/semantic_attributes/attribute_formats.rb [hunk 3]
```diff
@@ -44,12 +44,17 @@ def humanize(attr, obj)
...
         self.semantic_attributes[attr].predicates.inject(obj) { |val, predicate| val = predicate.normalize(val) }
       end
-      
+
       protected
 
       def define_normalization_method_for(attr)
-        self.define_attribute_methods if self.respond_to? :attribute_methods_generated? and !self.attribute_methods_generated?
-        
+        if Gem::Version.new(::ActiveRecord::VERSION::STRING) >= Gem::Version.new("4.0.4")
```
Comment (tiegz) [inline lib/semantic_attributes/attribute_formats.rb:51]: cc @ktheory

Comment (ktheory) [inline lib/semantic_attributes/attribute_formats.rb:51]: :+1:
